craft_parts/plugins/qmake_plugin.py

```diff
@@ -0,0 +1,145 @@
+        )
+        return cls(**plugin_data)
+
+
+class QmakePlugin(Plugin):
+    """The qmake plugin is useful for building qmake-based parts.
+
+    These are projects that are built using .pro files.
+
+    This plugin uses the common plugin keywords as well as those for "sources".
+    For more information check the 'plugins' topic for the former and the
+    'sources' topic for the latter.
+
+    Additionally, this plugin uses the following plugin-specific keywords:
+
+    - qmake-parameters:
+      (list of strings)
+      additional options to pass to the qmake invocation.
+
+    - qmake-project-file:
+      (string)
+      the qmake project file to use. This is usually only needed if
+      qmake can not determine what project file to use on its own.
+
+    - qmake_major_version:
+      (int)
+      set the qt major version. This is only needed to support qt6 builds.
+      Version 5 is default.
+    """
+
+    properties_class = QmakePluginProperties
+
+    def get_build_snaps(self) -> Set[str]:
+        """Return a set of required snaps to install in the build environment."""
+        return set()
+
+    def get_build_packages(self) -> Set[str]:
+        """Return a set of required packages to install in the build environment."""
+        options = cast(QmakePluginProperties, self._options)
+
+        if options.qmake_major_version == 6:
+            build_packages = {"g++", "make", "qmake6"}
+        else:
+            build_packages = {"g++", "make", "qt5-qmake"}
+        return build_packages
+
+    def get_build_environment(self) -> Dict[str, str]:
+        """Return a dictionary with the environment to use in the build step."""
+        options = cast(QmakePluginProperties, self._options)
+
+        if options.qmake_major_version == 6:
+            return {"QT_SELECT": "qt6"}
+
+        return {"QT_SELECT": "qt5"}
+
+    def get_build_commands(self) -> List[str]:
+        """Return a list of commands to run during the build step."""
+        options = cast(QmakePluginProperties, self._options)
+
+        if options.qmake_major_version == 6:
+            qmake_configure_command = [
+                "qmake6",
+                'QMAKE_CFLAGS+="${CFLAGS:-}"',
+                'QMAKE_CXXFLAGS+="${CXXFLAGS:-}"',
+                'QMAKE_LFLAGS+="${LDFLAGS:-}"',
+            ] + options.qmake_parameters
+        else:
+            qmake_configure_command = [
+                "qmake",
+                'QMAKE_CFLAGS+="${CFLAGS:-}"',
+                'QMAKE_CXXFLAGS+="${CXXFLAGS:-}"',
+                'QMAKE_LFLAGS+="${LDFLAGS:-}"',
+            ] + options.qmake_parameters
+
+        if options.qmake_project_file:
+            qmake_configure_command.extend(
+                [
+                    f'"{self._part_info.part_src_dir}"',
+                    "/",
+                    f'"{options.qmake_project_file}"',
+                ]
+            )
+        else:
+            qmake_configure_command.append(f'"{self._part_info.part_src_dir}"')
```

Comment:
Since this argument is the same in both cases, maybe we could add it before the conditional?